### ISSUE TITLE
[BUG] Website URL broken after subscribing a feed

### ISSUE THREAD

**Issue by math-GH** [FreshRSS/FreshRSS#4257]
I subscribed this feed: https://swimswam.com/deutsch/feed/

it sets the website URL to `<https://swimswam.com/wp-json/>; rel=https://api.w.org/, https://swimswam.com/wp-json/wp/v2/categories/43600; type=application/json`

![grafik](https://user-images.githubusercontent.com/1645099/156808151-9350e172-9e7a-4e3d-bb57-3c1cd77cf5b2.png)

**Expected behavior**
maybe an empty website URL

**Environment information (please complete the following information):**
- FreshRSS version: 1.20.0-dev

If the website URL field keeps empty than try it in development mode

**Comment by Frenzie** [comment]
I haven't looked at the code, but a quick glance in cURL suggests it could be a bug in SimplePie? That `link` header should have no bearing on anything except detecting feeds (while this *is* a feed), and even then `type="application/json"` is clearly not supported.

```
$ curl -I https://swimswam.com/deutsch/feed/
HTTP/2 200 
date: Fri, 04 Mar 2022 17:48:24 GMT
content-type: application/rss+xml; charset=UTF-8
vary: Accept-Encoding
vary: Accept-Encoding
x-gateway-request-id: d6aee461d9b5708015194435802a14d6
cf-edge-cache: cache,platform=wordpress
link: <https://swimswam.com/wp-json/>; rel="https://api.w.org/", <https://swimswam.com/wp-json/wp/v2/categories/43600>; rel="alternate"; type="application/json"
set-cookie: swpm_session=5dd8100cbdce5d20bd66a888eb84b035; path=/
last-modified: Fri, 04 Mar 2022 17:45:42 GMT
etag: W/"3b980e34540318d38b2626e22069ac18-gzip"
x-gateway-cache-key: 1646416044.945|standard|https|swimswam.com||/deutsch/feed/
x-gateway-cache-status: MISS
x-gateway-skip-cache: 0
cf-cache-status: DYNAMIC
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=tCnSshx3RDakIt7h3E3E%2BAmh6XU1xj0EG7zF%2FLswDSVFAYBEE%2BVrqRJB0DCYlnGO%2FGc%2BtP4Q9BgtJu0ylpcebyj0t7015JvbzXRjlw5WYhGM95GNZf8Q7RA97rnBkg%3D%3D"}],"group":"cf-nel","max_age":604800}
nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
server: cloudflare
cf-ray: 6e6c71495fe083cb-BRU
```

**Comment by math-GH** [comment]
@Frenzie  could you please paste the `curl` of `https://swimswam.com/feed/`? It is the same page, but it is the official feed (articles are in English).

**Comment by Frenzie** [comment]
```
$ curl -I https://swimswam.com/feed/
HTTP/2 200 
date: Fri, 04 Mar 2022 18:48:13 GMT
content-type: application/rss+xml; charset=UTF-8
vary: Accept-Encoding
vary: Accept-Encoding
x-gateway-request-id: cd273e5abafc35974ec444a02d5dbc78
cf-edge-cache: cache,platform=wordpress
link: <https://swimswam.com/wp-json/>; rel="https://api.w.org/"
last-modified: Fri, 04 Mar 2022 18:24:01 GMT
etag: W/"d2abf7c11e15763cb7c9e5028f2c8abf-gzip"
x-gateway-cache-key: 1646419492.781|standard|https|swimswam.com||/feed/
x-gateway-cache-status: HIT
x-gateway-skip-cache: 0
cf-cache-status: DYNAMIC
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=SKlQwmTdZ0CaQ%2B1jWKZB1FIFsAroOHpBvuyQyjkJuBUWinesgEEYkNos%2Bk5MIZpl8g109v29rHa5Hpst9wuIsYaWfFknzBzR3w7btdbqYOVCdUU%2B0jJck6wr0sUpng%3D%3D"}],"group":"cf-nel","max_age":604800}
nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
server: cloudflare
cf-ray: 6e6cc8e95ecc2e2c-BRU
```

**Comment by math-GH** [comment]
Will be fixed with #4283 
I checked it
